vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java

```diff
@@ -133,7 +135,48 @@ public CreationReturnValue createImageBlock(String slideId, byte[] image, String
         return returnValue;
     }
 
+    /**
+     * Delete a text block using an id
+     * 
+     * @param id - id of resource to be deleted. If the id is null then the
+     *           functions returns nothing.
+     *
+     */
+
     @Override
+    public void deleteTextBlockById(String id) throws BlockDoesNotExistException {
+        if (id == null) {
+            return;
+        }
+        try {
+            textBlockRepo.deleteById(id);         
+        } catch (EmptyResultDataAccessException e) {
+            throw new BlockDoesNotExistException(e);
+        }
+
+    }
+
+    /**
+     * Delete an image block using an id
+     * 
+     * @param id - id of resource to be deleted. If the id is null then the
+     *           functions returns nothing.
+     *
+     */
+
+    @Override
+    public void deleteImageBlockById(String id) throws BlockDoesNotExistException {
+        if (id == null) {
+            return;
+        }
+        try {
+            imageBlockRepo.deleteById(id);
+        } catch (IllegalArgumentException | EmptyResultDataAccessException e) {
```

Comment:
since id can't be null, IllegalArgumentException can't be thrown and for the other exception, I don't think it is thrown by this method?

Reply:
I removed IllegalArgumentException.